time-series/time-series-api/src/main/java/com/powsybl/timeseries/ast/NodeCalc.java

```diff
@@ -35,6 +37,16 @@ static void writeJson(NodeCalc node, JsonGenerator generator) {
         }
     }
 
+    static <R, A> R safeAccept(NodeCalc node, NodeCalcVisitor<R, A> visitor, A arg) {
+        Objects.requireNonNull(node);
+        Objects.requireNonNull(visitor);
+        try {
+            return node.accept(visitor, arg);
+        } catch (StackOverflowError error) {
+            throw new NodeCalcTooManyRecursionException(node, error);
+        }
+    }
```

Comment:
I’m not sure that it’s the best way to fix the issue: is it really the depth of the expression that causes this StackOverflow? I guess you’re facing to a cycle and I’m not convinced that is the proper way to fix this issue: this only catch the exception and give an understandable message, but this issue is that there is a cycle in the expression? Do I right?

Reply:
No it is not a matter of cycle, just a expression depth and a real stackoverflow. As there is no way to limit expressions size when writing a calculated time series, it can happen and the issue is that StackOverflowError is an Error and not catch everywhere which lead to hanging program...